cablelabs/transparent-security
# Fix GRE tunnel setup between network nodes and tofino switch VMs

#### What does this PR do?
Fixes #50 
Fixes GRE tunnel creation between VMs that act as network nodes and ones that act as Tofino switches
#### Do you have any concerns with this PR?
Still need more changes to get existing integration tests to work...
#### How can the reviewer verify this PR?
Run the different tofino automation scenarios and manually verify that the GRE tunnels are getting setup properly. One can either directly sniff the veth port in question or look at /var/log/transparent-security/model_0.log which will log the packets being processed.
#### Any background context you want to provide?
Need to get this working before we deploy to hardware as the P4 programs need to be changed more as some of the things we are doing in the BMV2 model are not compliant with tofino
#### Screenshots or logs (if appropriate)
#### Questions:
- Have you connected this PR to the issue it resolves? yes
- Does the documentation need an update? probably later after the tests are operational
- Does this add new dependencies? not really
- Have you added unit or functional tests for this PR? not at this time
